From a4c351f36f9232101d968740999310d3c1c24d89 Mon Sep 17 00:00:00 2001
From: Jayachandran B <jayachandran.b@intel.com>
Date: Wed, 3 Aug 2016 16:44:22 +0530
Subject: [PATCH 4691/4706] ASoC: Intel: Skylake: Probe - Fix function
 inlining issue

commit 1751366fd5daf57deb3325aa16e2b589a8cc9f97 from
git://git.yoctoproject.org/linux-yocto-4.1

get_hdac_ext_compr_stream was defined in a c file,
but was declared as inline in a header file.
This caused compilation issue when CONFIG_OPTIMIZE_INLINING
option was turned off.

Moved the function definition to the header file.

Change-Id: I0bd1900e6b30f8c86a335714d11a74faef1cd8ef
Signed-off-by: Jayachandran B <jayachandran.b@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/10744
Reviewed-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
Tested-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
---
 sound/soc/intel/skylake/skl-compress.c |    6 +-----
 sound/soc/intel/skylake/skl-compress.h |   17 ++++++++++++++---
 2 files changed, 15 insertions(+), 8 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-compress.c b/sound/soc/intel/skylake/skl-compress.c
index c8b26e8..de58eb8 100644
--- a/sound/soc/intel/skylake/skl-compress.c
+++ b/sound/soc/intel/skylake/skl-compress.c
@@ -25,11 +25,7 @@
 #include <sound/pcm_params.h>
 #include <sound/soc.h>
 #include "skl.h"
-inline
-struct hdac_ext_stream *get_hdac_ext_compr_stream(struct snd_compr_stream *stream)
-{
-	return stream->runtime->private_data;
-}
+#include "skl-compress.h"
 
 struct hdac_ext_bus *get_bus_compr_ctx(struct snd_compr_stream *substream)
 {
diff --git a/sound/soc/intel/skylake/skl-compress.h b/sound/soc/intel/skylake/skl-compress.h
index 9fcf6c38f..4e82b7a 100644
--- a/sound/soc/intel/skylake/skl-compress.h
+++ b/sound/soc/intel/skylake/skl-compress.h
@@ -19,8 +19,19 @@
  *
  */
 
-inline
-struct hdac_ext_stream *get_hdac_ext_compr_stream(struct snd_compr_stream *stream);
+#ifndef __SKL_COMPRESS_H__
+#define __SKL_COMPRESS_H__
+
+#include <sound/soc.h>
+#include <sound/memalloc.h>
+#include "skl.h"
+
+static inline
+struct hdac_ext_stream *get_hdac_ext_compr_stream(struct snd_compr_stream *stream)
+{
+	return stream->runtime->private_data;
+}
+
 struct hdac_ext_bus *get_bus_compr_ctx(struct snd_compr_stream *substream);
 void skl_set_compr_runtime_buffer(struct snd_compr_stream *substream,
 				struct snd_dma_buffer *bufp, size_t size);
@@ -32,4 +43,4 @@ int skl_substream_alloc_compr_pages(struct hdac_ext_bus *ebus,
 int skl_compr_free_pages(struct snd_compr_stream *substream);
 int skl_substream_free_compr_pages(struct hdac_bus *bus,
 				struct snd_compr_stream *substream);
-
+#endif
-- 
1.7.5.4

